### PR TITLE
added contract negotiation demo logic to http communication

### PR DESCRIPTION
**Pull request by davidjovanovic**

### CODE REVIEW

File: src/main/resources/config.properties
Comment (marest94) [inline src/main/resources/config.properties:1]: Remove your local path to ssl keystore :)

File: src/main/resources/application.properties
Comment (marest94) [inline src/main/resources/application.properties:1]: Why fileSenderPort is changed?

Comment (davidjovanovic) [reply src/main/resources/application.properties:1]: I remebered it being like this always. I was surprised that it was changed to 9000 for the sender :smile:

File: src/main/resources/config.properties
Comment (davidjovanovic) [reply src/main/resources/config.properties:1]: oooops :smile:  big thanks!

File: src/main/java/it/eng/idsa/dataapp/service/impl/SelfDescriptionServiceImpl.java [hunk 1]
```diff
@@ -40,27 +39,21 @@ public class SelfDescriptionServiceImpl implements SelfDescriptionService {
...
 
-		int eccPort = (Boolean.TRUE.equals(((Boolean) threadService.getThreadLocalValue("wss"))))
-				? eccProperties.getWssSelfDescriptionPort()
-				: eccProperties.getPort();
 		try {
-			eccURI = new URI(eccProperties.getProtocol(), null, eccProperties.getHost(), eccPort,
-					eccProperties.getSelfdescriptionContext(), null, null);
-			logger.info("Fetching self description from ECC {}.", eccURI.toString());
+			eccURI = new URI(eccProperties.getProtocol(), null, eccProperties.getHost(), eccProperties.getSelfdescriptionPort(),
+					"/internal/sd", null, null);
```
Comment (IgorBalog-Eng) [inline src/main/java/it/eng/idsa/dataapp/service/impl/SelfDescriptionServiceImpl.java:55]: Why not extracted to constant at the declaration part? :)

File: src/main/java/it/eng/idsa/dataapp/service/impl/SelfDescriptionServiceImpl.java [hunk 2]
```diff
@@ -40,25 +39,19 @@ public class SelfDescriptionServiceImpl implements SelfDescriptionService {
...
 		this.eccProperties = eccProperties;
-		this.threadService = threadService;
 
 	}
 
 	@Override
 	public Connector getSelfDescription(Message message) {
 		URI eccURI = null;
 
-		int eccPort = (Boolean.TRUE.equals(((Boolean) threadService.getThreadLocalValue("wss"))))
```
[the remark this answers is not in the document]
Comment (davidjovanovic) [reply src/main/java/it/eng/idsa/dataapp/service/impl/SelfDescriptionServiceImpl.java:57]: :+1:

File: src/main/java/it/eng/idsa/dataapp/service/impl/SelfDescriptionServiceImpl.java (hunk 1 above)
Comment (davidjovanovic) [reply src/main/java/it/eng/idsa/dataapp/service/impl/SelfDescriptionServiceImpl.java:55]: I'll do it right away :)

**Pull request merged**